gmakefile.test

```diff
@@ -263,19 +263,20 @@ else ifdef searchin
 else ifdef argsearch
   TESTTARGETS := $(foreach v,$(alltesttargets),$(if $(findstring $(argsearch),$($(v)_ARGS)),$(v)))
 else ifdef globsearch
-  TESTTARGETS := $(shell $(PYTHON) -c"import sys,fnmatch,itertools; m=[fnmatch.filter(sys.argv[2].split(),p) for p in sys.argv[1].split()]; print(' '.join(list(itertools.chain.from_iterable(m))))" '$(globsearch)' '$(alltesttargets)')
+  TESTTARGETS := $(shell $(PYTHON) $(PETSCCONFIGDIR)/query_tests.py --testdir=$(TESTDIR) --srcdir=$(TESTSRCDIR) 'name' '$(globsearch)')
```

Comment:
I am bit reluctant to unnecessary break the functionality with 3.13.0 when the original solution will work perfectly OK (for us) even with 3.13.1. On the other hand, I don't think the globsearch is actually really useful (for us).

Reply:
I would be reluctant if it was a change in master but it's a change in maint/patch release. This approach simpler and keeps pace with PETSc. Yes, globsearch is probably not that much used - the more so we can take the newer approach, because probability that somebody would need it with older PETSc version is ~0.